internal/jennies/terraform/tools.go

```diff
@@ -0,0 +1,56 @@
+package terraform
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/grafana/cog/internal/tools"
+)
+
+func formatPackageName(pkg string) string {
+	rgx := regexp.MustCompile("[^a-zA-Z0-9_]+")
+
+	return strings.ToLower(rgx.ReplaceAllString(pkg, ""))
+}
+
+func formatArgName(name string) string {
+	return escapeVarName(tools.LowerCamelCase(name))
+}
+
+func escapeVarName(varName string) string {
+	if isReservedGoKeyword(varName) {
+		return varName + "Arg"
+	}
+
+	return varName
+}
+
+func isReservedGoKeyword(input string) bool {
```

Comment:
Lots of unused functions. I'll review once it's been cleaned up 😄

Reply:
Cleaned up. Far from complete, but at least generates models (and their scalar fields)

Reply:
First I'm gonna make what it creates compile. Then I'll get some tests working. And perhaps document how you can run it for yourself.

Reply:
The code it creates now compiles (I temporarily excluded anything in the model schema that isn't a scalar) and I've formatted names better, so no `DashboardDashboardDatasourceModel`.

Reply:
I just committed some test data that should show you how this renders.